### DISCUSSION TITLE
Modify Img file

### DISCUSSION THREAD

**Discussion by thierryforb** [PartialVolume/shredos.x86_64#130] [General]
Hello,
Is there a way, for modifying the img file before to copy it on a usb stick ?

I've tried to:
mount -o loop,offset=646656 shredos-2021.08.2_23_x86-64_0.34_20221231.img mount/
Then, modify grub.cfg in the boot, and EFI directory (as said in the documentation)
umount /mount
and finally dd if=shredos-2021.08.2_23_x86-64_0.34_20221231.img of=/dev/sda1
But it looks like the modification brooked grub
The usb stick, does not boot and I get only a grub prompt.
Is there anything else that i can do before doing the dd ?
Thanks a lot for your help

**Comment by PartialVolume** [comment]
I've never tried doing it like that. I've no idea if that would work.

This is the way I would do it. Download the source then modify ../board/shredos/grub.cfg then build it as described [here](https://github.com/PartialVolume/shredos.x86_64#compiling-shredos-and-burning-to-usb-stick-the-harder-way-)

**Comment by PartialVolume** [comment]
I guess another way, that's similar to how you are trying to do it would be to dd the .img file to a USB stick, edit both grub.cfg files on the USB stick then dd the USB stick back to a modified.img file. To keep the .img file small rather than the size of the USB stick you would need to figure out where the vfat partition ends and tell dd to only copy upto that point. I've never tried it but it should work.

**Comment by thierryforb** [comment]
Thanks for the answers, 
I've more questions, what is the filesystem used by shredos ? my ubuntu don't recognize it, gparted neither...
A strange thing is that when i convert the img that I've modified, to a vdi it perfectly boots in Virtualbox, via this command : 
vboxmanage convertfromraw --format VDI shredos-2021.08.2_23_x86-64_0.34_20221231.img shredos-2021.08.2_23_x86-64_0.34_20221231.vdi
Then I tried to convert it back to img via : VBoxManage clonemedium --format RAW shredos-2021.08.2_23_x86-64_0.34_20221231.vdi test.img
But after a dd I got the same problem

**Reply by PartialVolume** [reply]
You mean the file system that ends up on the USB stick? That's vfat. To mount it just use `mkdir tmpmnt;sudo mount /dev/sdx tmpmnt` or `sudo mount -t vfat /dev/sdx tmpmnt` works too then to umount `sudo umount tmpmnt`

fdisk -l shows it as W95 FAT32 (LBA)
>Device     Boot Start      End  Sectors  Size Id Type
/dev/sdf1        1263 60063743 60062481 28.6G  c W95 FAT32 (LBA) 

Recognised in gparted on my system:
![Screenshot_20230306_092537](https://user-images.githubusercontent.com/22084881/223070046-5122f23d-4d0f-4488-8baa-dec750f7acf0.png)
